content/articles/naive-bayes-algorithm-in-python/index.md

```diff
@@ -0,0 +1,141 @@
+# Importing the dataset
+data = read.csv('data.csv')
+# Looking at the first 10 observationd of our dataset
+head(data)
+
+```
+#### Output
+
+![image](/engineering-education/naive-bayes-algorithm-in-python/output-image.png)
+
+As we can see, the data has two features, i.e., `Age` and `EstimatedSalary` with the Purchased as the target variable, which takes value 1 for customer bought and 0 for customer didn't buy the product. On the rows is the number corresponding to a particular customer. Hoping we now understand our data better. Let's continue with our preprocessing activity. The code snippet below concludes this activity.
+
+
+```r
+# Encoding the target variable
+data$Purchased = factor(data$Purchased, levels = c(0, 1))
+# Splitting the data into Training-Test sets
+library(caTools)
+set.seed(123)
+split = sample.split(data$Purchased, SplitRatio = 0.80)
+Train_set = subset(data, split == TRUE)
+Test_set = subset(dataset, split == FALSE)
+
+# Feature Scaling
+Train_set[-3] = scale(Train_set[-3])
+Test_set[-3] = scale(Test_set[-3])
+
+```
+### Step 2: Fitting the Naive Bayes classifier to the Training set
+To get started, let's make sure that the `e1071 package` is installed on R. As we earlier said, we only install this package if it is not already installed in on our systems; otherwise, we proceed and load its library.
+
+```r
+library(e1071) # load the library
+classifier = naiveBayes(x = Train_set[-3],
+                        y = Train_set$Purchased) # Fits Naive Bayes Model to the training set
+
+```
+
+### Step 3: Predicting the Test set results
+
+```r
+# Predicting the Test set outputs
+y_predict = predict(classifier, newdata = Test_set[-3])
+
+# Creating a Confusion Matrix
+conf_matrx = table(Test_set[, 3], y_predict)
+conf_matrx
+```
+#### Obtained Confusion Matrix
+
+![confusion matrix](/engineering-education/naive-bayes-algorithm-in-python/confusion-matrix.png)
+
+From the above confusion matrix, we notice that out of 100, the model could predict 86% of the data correctly with only 14% incorrect predictions. From this, it's clear that our model has an accuracy of 84%. The accuracy of 84% is a good score, and thus we can conclude that our classifier was able to classify our data accurately.
+
+### Conclusion
+In this tutorial, we have learned the Naive Bayes classifier's theory. First, we showed how to derive a mathematical formula of this classifier from the basic conditional probability. Later, we showed how to implement the  Naive Bayes classifier in R and evaluated its performance using a confusion matrix. From the confusion matrix, we saw its ability to classify the data by giving a relatively incredible score. Since we now understand this classifier better, we can widen our knowledge by challenging ourselves with more implementation tasks to handle problems such as email classification, transaction classification, and health data to classify tumours and other diseases. Here we reach the end of this session's learning.
+
```

Comment:
```suggestion
In this tutorial, we have learned the Naive Bayes classifier's theory. First, we showed how to derive a mathematical formula of this classifier from the basic conditional probability. Later, we showed how to implement the  Naive Bayes classifier in R and evaluated its performance using a confusion matrix. From the confusion matrix, we saw its ability to classify the data by giving a relatively incredible score. Since we now understand this classifier better, we can widen our knowledge by challenging ourselves with more implementation tasks to handle problems such as email classification, transaction classification, and health data to classify tumors and other diseases. Here we reach the end of this session's learning.
```